core/src/test/java/kafka/server/share/DelayedShareFetchTest.java

```diff
@@ -256,21 +252,26 @@ public void testTryCompleteWhenMinBytesNotSatisfiedOnFirstFetch() {
         when(sp0.maybeAcquireFetchLock(fetchId)).thenReturn(true);
         when(sp1.maybeAcquireFetchLock(fetchId)).thenReturn(true);
 
-        assertFalse(delayedShareFetch.isCompleted());
+        try (MockedStatic<ShareFetchUtils> mockedShareFetchUtils = Mockito.mockStatic(ShareFetchUtils.class, Mockito.CALLS_REAL_METHODS)) {
+            mockedShareFetchUtils.when(() -> ShareFetchUtils.processFetchResponse(any(), any(), any(), any(), any()))
+                    .thenReturn(Map.of(tp0, mock(ShareFetchResponseData.PartitionData.class)));
 
-        // Since sp1 cannot be acquired, tryComplete should return false.
-        assertFalse(delayedShareFetch.tryComplete());
-        assertFalse(delayedShareFetch.isCompleted());
-        Mockito.verify(delayedShareFetch, times(1)).releasePartitionLocks(any());
-        assertTrue(delayedShareFetch.lock().tryLock());
-        // Though the request is not completed but sp0 was acquired and hence the metric should be recorded.
-        assertEquals(1, shareGroupMetrics.topicPartitionsAcquireTimeMs(groupId).count());
-        assertEquals(10, shareGroupMetrics.topicPartitionsAcquireTimeMs(groupId).sum());
-        // Since the request is not completed, the fetch ratio should be null.
-        assertNull(shareGroupMetrics.topicPartitionsFetchRatio(groupId));
+            assertFalse(delayedShareFetch.isCompleted());
 
-        delayedShareFetch.lock().unlock();
-        Mockito.verify(exceptionHandler, times(1)).accept(any(), any());
+            // Since minBytes(2) is not satisfied (only 1 byte available from sp0), and sp1 cannot be acquired, tryComplete should return false.
+            assertFalse(delayedShareFetch.tryComplete());
+            assertFalse(delayedShareFetch.isCompleted());
+            Mockito.verify(delayedShareFetch, times(1)).releasePartitionLocks(any());
+            assertTrue(delayedShareFetch.lock().tryLock());
+            // Though the request is not completed but sp0 was acquired and hence the metric should be recorded.
+            assertEquals(1, shareGroupMetrics.topicPartitionsAcquireTimeMs(groupId).count());
+            assertEquals(10, shareGroupMetrics.topicPartitionsAcquireTimeMs(groupId).sum());
+            // Since the request is not completed, the fetch ratio should be null.
+            assertNull(shareGroupMetrics.topicPartitionsFetchRatio(groupId));
+
+            delayedShareFetch.lock().unlock();
+            Mockito.verify(exceptionHandler, never()).accept(any(), any());
```

Comment:
The exceptionHandler shouldn't be triggered in this case. 
The original test case has exceptions since https://github.com/apache/kafka/blob/trunk/core/src/test/java/kafka/server/share/DelayedShareFetchTest.java#L238 override the return value in `replicaManager.getPartitionOrException(tp0.topicPartition())` defined in https://github.com/apache/kafka/blob/trunk/core/src/test/java/kafka/server/share/DelayedShareFetchTest.java#L225
which cause NPE in https://github.com/apache/kafka/blob/2503c742aea63b8c0584fe60559726ac6586a67c/core/src/main/java/kafka/server/share/DelayedShareFetch.java#L481, this is unexpected.